src/Jackalope/Transport/DoctrineDBAL/Client.php

```diff
@@ -1539,8 +1539,13 @@ private function getSystemIdForNode($identifier, $workspaceName = null)
         if (UUIDHelper::isUUID($identifier)) {
             $query = 'SELECT id FROM phpcr_nodes WHERE identifier = ? AND workspace_name = ?';
         } else {
-            if ($this->getConnection()->getDriver() instanceof \Doctrine\DBAL\Driver\PDOMySql\Driver) {
-                $query = 'SELECT id FROM phpcr_nodes WHERE path COLLATE utf8_bin = ? AND workspace_name = ?';
+            $params = $this->conn->getParams();
+            $charset = isset($params['charset']) ? $params['charset'] : 'utf8';
```

Comment:
hmm would need to dig in here too .. but I agree we should do as little work as possible at run time ..
so what we can do at install time or at config compile time the better.